### DISCUSSION TITLE
Proper way handle different auth

### DISCUSSION THREAD

**Discussion by harryqt** [saloonphp/saloon#232] [Q&A] [answered]
I have an api, where some endpoints require "x-api-key" based auth, some require JWT based auth, some don't require any.

I'm trying to figure out how I should structure them, any recommendation?

**Comment by Sammyjo20** [comment, marked as answer]
Hey @Dibbyo456 

You could pass in the requirements on a per-request basis into the constructor of the request and then setting the headers based on the request. 

```php
class SomeRequest extends Request
{
     public function __construct(string $apiKey)
     {
            $this->headers()->add('X-API-KEY', $apiKey);
     }
}
```

You could also make your own [custom authenticators](https://docs.saloon.dev/the-basics/authentication#custom-authenticators) if you have multiple routes that require JWT/X-API-KEY 

```php
class SomeRequest extends Request
{
     protected function defaultAuth()
     {
           return new JwtAuthenticator($this->token);
     }
}
```
